src/index.ts

```diff
@@ -195,9 +197,7 @@ export async function createWithExpiration (peerId: PeerId, value: CID | PeerId
 export async function createWithExpiration (peerId: PeerId, value: CID | PeerId | string, seq: number | bigint, expiration: string, options: CreateOptions = defaultCreateOptions): Promise<IPNSRecord> {
   const expirationDate = NanoDate.fromString(expiration)
   const validityType = IpnsEntry.ValidityType.EOL
-
-  const ttlMs = expirationDate.toDate().getTime() - Date.now()
-  const ttlNs = (BigInt(ttlMs) * BigInt(100000)) + BigInt(expirationDate.getNano())
+  const ttlNs = typeof options.ttlNs !== 'undefined' ? BigInt(options.ttlNs) : DEFAULT_TTL_NS
```

Comment:
`DEFAULT_TTL_NS` is a `number` but we need it to be a `bigint`.

```suggestion
  const ttlNs = BigInt(options.ttlNs != null ? options.ttlNs : DEFAULT_TTL_NS)
```